### PR TITLE
Update neo4j check to use requests wrapper

### PR DESCRIPTION
**Pull request by npezzotti**
### What does this PR do?

- Updates neo4j to use RequestsWrapper
- Adds `spec.yaml`
- Simplifies dd_environment by setting the test instance password in the docker-compose file

### Motivation

- Standardize HTTP requests
- Reduce complexity in test scenario

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: neo4j/datadog_checks/neo4j/neo4j.py [hunk 1]
```diff
@@ -86,11 +92,10 @@ class Neo4jCheck(AgentCheck):
     }
 
     def check(self, instance):
```
Comment (coignetp) [inline neo4j/datadog_checks/neo4j/neo4j.py:94]:
```suggestion
    def check(self, _):
```

File: neo4j/datadog_checks/neo4j/neo4j.py [hunk 2]
```diff
@@ -86,11 +92,10 @@ class Neo4jCheck(AgentCheck):
...
-        host, port, user, password, timeout, server_name = self._get_config(instance)
+        host, port, server_name = self._get_config(instance)
```
Comment (coignetp) [inline neo4j/datadog_checks/neo4j/neo4j.py:95]:
```suggestion
        host, port, server_name = self._get_config(self.instance)
```

File: neo4j/tests/conftest.py [hunk 3]
```diff
@@ -27,6 +18,5 @@ def dd_environment():
         os.path.join(DOCKER_DIR, 'docker-compose.yaml'),
         env_vars=envs,
```
Comment (coignetp) [inline neo4j/tests/conftest.py:19]: Let's add `NEO4J_AUTH` env variable here and not in the docker-compose

File: neo4j/tests/docker/docker-compose.yaml [hunk 4]
```diff
@@ -5,3 +5,5 @@ services:
     image: neo4j:${NEO4J_VERSION}
     ports:
       - 7474:7474
+    environment: 
+      NEO4J_AUTH: neo4j/dog
```
Comment (coignetp) [inline neo4j/tests/docker/docker-compose.yaml:9]:
```suggestion
```

File: neo4j/assets/configuration/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    options:
+    - name: neo4j_url
+      required: true
+      description: |
+        Neo4J url to connect to.
+      value:
+        type: string
+        example: localhost
+    - name: port
+      required: true
```
Comment (coignetp) [inline neo4j/assets/configuration/spec.yaml:19]:
```suggestion
```
It looks optional in the code

File: neo4j/datadog_checks/neo4j/data/conf.yaml.example [hunk 6]
```diff
@@ -1,53 +1,373 @@
...
+    server_name: <SERVER_NAME>
 
-    ## @param password - string - required
-    ## Password associated with the `user` to connect to the
-    ## Neo4j instance
+    ## @param neo4j_version - string - required
+    ## The Neo4j version to gather metric from.
     #
-    password: <PASSWORD>
+    neo4j_version: <NEO4J_VERSION>
```
Comment (coignetp) [inline neo4j/datadog_checks/neo4j/data/conf.yaml.example:66]: Is it a config option?

**Review by coignetp — CHANGES_REQUESTED** [review]

File: neo4j/datadog_checks/neo4j/data/conf.yaml.example (hunk 6 above)
Comment (npezzotti) [reply neo4j/datadog_checks/neo4j/data/conf.yaml.example:66]: Ah nice catch- yes it doesn't appear to be used at all, I'll remove it.

**Review by coignetp — APPROVED** [review]

**Pull request merged**